src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesVsPatterns.cs

```diff
@@ -2125,5 +2125,119 @@ void M(object? p)
 ");
             c.VerifyDiagnostics();
         }
+
+        [Fact]
+        public void NotNullIsAPureNullTest()
+        {
+            var source =
+@"#nullable enable
+class C
+{
+    void M1(C? x)
+    {
+        if (x is not null)
+            x.ToString();
+        x.ToString(); // 1
+    }
+    void M2(C x)
+    {
+        if (x is not null)
+            x.ToString();
+        x.ToString(); // 2
+    }
+    void M3(C x)
+    {
+        if (x is not null or _)
+            x.ToString(); // 3
+        x.ToString();
+    }
+    void M4(C x)
+    {
+        if (x is null or _)
+            x.ToString(); // 4
+        x.ToString();
```

Comment:
Would we expect a warning here if the 'if' had an empty body?

Reply:
Yes.